### PR TITLE
Implement pg_get_expr() for subpartition template

### PR DESCRIPTION
**Pull request by l-wang**
Co-authored-by: Ashwin Agrawal <aashwin@vmware.com>

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/test/regress/expected/gp_partition_template.out [hunk 1]
```diff
@@ -445,10 +445,10 @@ create table subpart_templ_encoding (i int, j int, k int, l int)
...
+SELECT level, pg_get_expr(template, relid) from gp_partition_template t WHERE t.relid = 'subpart_templ_encoding'::regclass;
+ level |                                                                                                                        pg_get_expr                                                                                                                         
+-------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+     1 | SUBPARTITION TEMPLATE(SUBPARTITION sp1 START (100::integer) END (200::integer) Every (50), COLUMN i ENCODING (compresstype=zlib), COLUMN j ENCODING (compresstype=rle_type), COLUMN k ENCODING (compresstype=zlib), COLUMN l ENCODING (compresstype=zlib))
```
Comment (divyeshddv) [inline src/test/regress/expected/gp_partition_template.out:451]: Can we also add a test for 'default column encoding' case 
or having default encoding options passed using the WITH clause

File: src/test/regress/expected/partition_optimizer.out [hunk 2]
```diff
@@ -2347,28 +2347,28 @@ select relid::regclass, level from gp_partition_template where relid = 'rank_set
...
- rank_settemp |     1 | {GPPARTITIONDEFINITION :partDefElems ({GPPARTDEFELEM :partName <> :boundSpec {GPPARTITIONRANGESPEC :partStart {GPPARTITIONRANGEITEM :val ({CONST :consttype 1082 :consttypmod -1 :constcollid 0 :constlen 4 :constbyval true :constisnull false :location -1 :constvalue 4 [ -3 9 0 0 0 0 0 0 ]}) :edge 1} :partEnd <> :partEvery <>} :subSpec <> :isDefault false :options ({DEFELEM :defnamespace <> :defname compresslevel :arg 5 :defaction 0 :location 101}) :accessMethod ao_row :tablespacename <> :colencs <>}) :encClauses <> :isTemplate true}
+select relid::regclass, level, pg_get_expr(template, relid) from gp_partition_template where relid = 'rank_settemp'::regclass;
+    relid     | level |                    pg_get_expr                     
+--------------+-------+----------------------------------------------------
+ rank_settemp |     1 | SUBPARTITION TEMPLATE( START ('01-01-2007'::date))
```
Comment (divyeshddv) [inline src/test/regress/expected/partition_optimizer.out:2371]: How do we store the partition specific options? in this case `compresslevel=5` 

Also how do we store the partition's access method here? is it done in a different place/ or do we not need to?

File: src/test/regress/expected/gp_partition_template.out (hunk 1 above)
Comment (l-wang) [reply src/test/regress/expected/gp_partition_template.out:451]: Yes! I'm adding test case for both! I actually missed printing the WITH clause, thank you for the great catch!

File: src/test/regress/expected/partition_optimizer.out (hunk 2 above)
Comment (l-wang) [reply src/test/regress/expected/partition_optimizer.out:2371]: No the current syntax doesn't allow template to specify partition specific options. So pg_get_expr() doesn't expect to see options inside a partition spec, hence we don't need to print it.

**Review by divyeshddv — APPROVED** [review]
LGTM! Nice test coverage for these scenarios!

**Comment by l-wang** [comment]
Thank you @divyeshddv for reviewing!

**Pull request merged**